amplitude.py

```diff
@@ -20,18 +21,25 @@
 # specifying wbits=0 but that didn't work for me locally. This did.
 ZLIB_HEADER_SIZE = 32
 
-def handle (message):
+def handle (message, context):
     # http://docs.aws.amazon.com/AmazonS3/latest/dev/notification-content-structure.html
-    records = json.loads(message)["Records"]
+    if (type(message) is str):
+        message = json.loads(message)
+    records = message["Records"]
+
     for record in records:
         if record["eventSource"] != "aws:s3":
             continue
 
+        print record["s3"]["bucket"]["name"], record["s3"]["object"]["key"]
         s3 = boto3.resource("s3", region_name=record["awsRegion"])
         s3_object = s3.Object(record["s3"]["bucket"]["name"], record["s3"]["object"]["key"])
+        tmpfileobj, tmppath = tempfile.mkstemp()
+        s3_object.download_file(tmppath)
```

Comment:
If iterating on the s3 object directly doesn't work, we should still be able to process it without writing to the local file system I think.

@jbuck, I've pushed a proposed tweak in 02344ba, can you give it a whirl and report back?